ElasticsearchReportHelper.php

```diff
@@ -653,6 +654,54 @@ public static function leafletMap(array $options) {
     return self::getControlContainer('leafletMap', $options, $dataOptions);
   }
 
+  /**
+   * Output a selector for a survey.
+   *
+   * @return string
+   *   Select HTML.
+   *
+   * @link https://indicia-docs.readthedocs.io/en/latest/site-building/iform/helpers/elasticsearch-report-helper.html#elasticsearchreporthelper-surveyFilter
+   */
+  public static function surveyFilter(array $options) {
+    $options = array_merge([
+      'id' => "es-survey-filter",
+      'label' => lang::get('Limit to survey'),
+    ], $options);
+
+    iform_load_helpers(['report_helper']);
+    $sources = report_helper::get_report_data(array(
+      'dataSource' => 'library/surveys/surveys_list',
+      'readAuth' => $options['readAuth'],
+      'caching' => TRUE,
+      'cachePerUser' => FALSE,
+    ));
+
+    $optionArr = [];
+    $optionArr['all'] = lang::get('All surveys');
+    usort($sources, function ($item1, $item2) {
```

Comment:
Can sorting not be done in the get_report_data call (or even a get_population_data call which goes straight to data services rather than needing a report)? But, see below first for alternative suggestion.